### PR TITLE
D8 nid 1094 cold weather payments form

### PR DESCRIPTION
**Pull request by neilblair**
Enable the form to accept a full NI postcode plus changes to theme it up and add styled error message.

Screenshot 2021-02-18 at 16 00 42

### CODE REVIEW

File: nidirect_cold_weather_payments/src/Form/ColdWeatherPaymentCheckerForm.php [hunk 1]
```diff
@@ -224,4 +252,28 @@ private function cwpLookup($postcode) {
...
+   * Get postcode district from a Northern Ireland postcode.
+   *
+   * For Northern Ireland postcodes, the postcode district is always 1 or 2 digits following the postcode area 'BT'.
+   */
+  private function cwpGetPostcodeDistrict(string $postcode) {
+    $postcode_district = NULL;
+
+    // If postcode is a full NI postcode, or just the first part (outward code - e.g. BT1) ...
+    if (preg_match('/^[bB][tT][0-9]{1,2}( ?[0-9][a-zA-Z]{2})?$/', $postcode)) {
+      if (strlen($postcode) > 4) {
```
Comment (johangant) [inline nidirect_cold_weather_payments/src/Form/ColdWeatherPaymentCheckerForm.php:265]: I'd recommend you trim any spaces in `$postcode` here as a padding space between the two halves can affect your string length check and potentially some of the substring trim points could be offset by 1.

**Review by johangant — CHANGES_REQUESTED** [review]
Looks good. A space removal in `cwpGetPostcodeDistrict` would help avoid any potential offset issues with trimming the string to calculate the district code.

File: nidirect_cold_weather_payments/src/Form/ColdWeatherPaymentCheckerForm.php (hunk 1 above)
Comment (omahm) [reply nidirect_cold_weather_payments/src/Form/ColdWeatherPaymentCheckerForm.php:265]: I'd also consider lowercasing in the first instance so you can reduce the regex expression.

Comment (neilblair) [reply nidirect_cold_weather_payments/src/Form/ColdWeatherPaymentCheckerForm.php:274]: @johangant , @omahm  - see latest commit.

**Review by johangant — APPROVED** [review]
👍  nice one

**Pull request merged**